core/common/src/main/java/alluxio/security/LoginUser.java

```diff
@@ -50,18 +50,21 @@ private LoginUser() {} // prevent instantiation
    * @return the login user
    * @throws IOException if login fails
    */
-  public static User get() throws IOException {
+  public static synchronized User get() throws IOException {
     if (sLoginUser == null) {
-      synchronized (LoginUser.class) {
-        if (sLoginUser == null) {
-          sLoginUser = login();
-        }
-      }
+      sLoginUser = login();
     }
     return sLoginUser;
   }
 
   /**
+   * Resets the login user so that the next call to get() will have to log in again.
+   */
+  public static synchronized void reset() {
```

Comment:
I've moved the Whitebox manipulation to a util a time where we can break up the LoginUser singleton.